dns_sd_windows.c

```diff
@@ -19,42 +19,60 @@
 #include "iio-private.h"
 #include "mdns.h"
 
-static int new_discovery_data(struct dns_sd_discovery_data** data)
+#ifdef HAVE_IPV6
+static const unsigned char localhost[] = {
+	0, 0, 0, 0, 0, 0, 0, 0,
+	0, 0, 0, 0, 0, 0, 0, 1
+};
+static const unsigned char localhost_mapped[] = {
+	0, 0, 0,    0,    0,    0, 0, 0,
+	0, 0, 0xff, 0xff, 0x7f, 0, 0, 1
+};
+
+static inline bool is_localhost6(struct sockaddr_in6 *saddr6)
 {
-	struct dns_sd_discovery_data* d;
-
-	d = zalloc(sizeof(struct dns_sd_discovery_data));
-	if (!d)
-		return -ENOMEM;
+	const uint8_t *addr = saddr6->sin6_addr.s6_addr;
 
-	*data = d;
-	return 0;
+	return !memcmp(addr, localhost, sizeof(localhost)) ||
+		!memcmp(addr, localhost_mapped, sizeof(localhost_mapped));
 }
+#endif
 
-static int
-open_client_sockets(int* sockets, int max_sockets) {
-	// When sending, each socket can only send to one network interface
-	// Thus we need to open one socket for each interface and address family
-	int num_sockets = 0;
+static inline bool is_localhost4(struct sockaddr_in *saddr)
+{
+	return saddr->sin_addr.S_un.S_un_b.s_b1 == 127 &&
+		saddr->sin_addr.S_un.S_un_b.s_b2 == 0 &&
+		saddr->sin_addr.S_un.S_un_b.s_b3 == 0 &&
+		saddr->sin_addr.S_un.S_un_b.s_b4 == 1;
+}
```

Comment:
maybe make *saddr `const`?

Reply:
I feel like Codacity should report stuff like this :(

Reply:
Agreed. It spills so much stuff that you do not really care and this is something easy for us to forget but very important to have...